golang tip broken! ci-kubernetes-unit-golang-tip is failing

### Which jobs are failing?

https://prow.k8s.io/job-history/gs/kubernetes-ci-logs/logs/ci-kubernetes-unit-golang-tip

### Which tests are failing?

```
=== RUN   TestCheckpointStateRestore/PodLevelResourceManagers=false/Fail_to_restore_checkpoint_from_v1_(migration)
    state_checkpoint_test.go:532: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go:532
        	Error:      	Error "could not restore state from checkpoint: json: cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries.containerID1 of type map[string]string, please drain this node and delete the CPU manager checkpoint file \"/tmp/cpumanager_state_test949231204/cpumanager_checkpoint_test\" before restarting Kubelet" does not contain "cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries of type map[string]string"
        	Test:       	TestCheckpointStateRestore/PodLevelResourceManagers=false/Fail_to_restore_checkpoint_from_v1_(migration)
--- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=false/Fail_to_restore_checkpoint_from_v1_(migration) (0.00s)

=== RUN   TestCheckpointStateRestore/PodLevelResourceManagers=false
--- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=false (0.02s)

=== RUN   TestCheckpointStateRestore/PodLevelResourceManagers=true/Fail_to_restore_checkpoint_from_v1_(migration)
    state_checkpoint_test.go:532: 
        	Error Trace:	/home/prow/go/src/k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go:532
        	Error:      	Error "could not restore state from checkpoint: json: cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries.containerID1 of type map[string]string, please drain this node and delete the CPU manager checkpoint file \"/tmp/cpumanager_state_test949231204/cpumanager_checkpoint_test\" before restarting Kubelet" does not contain "cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries of type map[string]string"
        	Test:       	TestCheckpointStateRestore/PodLevelResourceManagers=true/Fail_to_restore_checkpoint_from_v1_(migration)
--- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=true/Fail_to_restore_checkpoint_from_v1_(migration) (0.00s)

=== RUN   TestCheckpointStateRestore/PodLevelResourceManagers=true
--- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=true (0.02s)

=== RUN   TestCheckpointStateRestore
--- FAIL: TestCheckpointStateRestore (0.04s)
```

### Since when has it been failing?

Above jobs have been failing since the merge of https://github.com/kubernetes/kubernetes/pull/139068

### Testgrid link

https://testgrid.k8s.io/sig-arch-code-organization#unit-master-golang-tip

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

I could locally confirm that this test started failing only after the above mentioned change: (This PASSES when run with stable go)
```
[root@raji-x86-workspace1 kubernetes]# git checkout c0a6b50f7f83caef1c58a803c3585fdd52157a27
Previous HEAD position was c07cd3400b6 Merge pull request #139636 from cheftako/anp-release
HEAD is now at c0a6b50f7f8 Merge pull request #139567 from serathius/watchcache-benchmark-not-older-than
[root@raji-x86-workspace1 kubernetes]# gotip test k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state -count=1 -run TestCheckpointStateRestore
ok  	k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state	0.336s
[root@raji-x86-workspace1 kubernetes]# git checkout 359baf1d4d169378f9357797e2fa1a02d854d34a
Previous HEAD position was c0a6b50f7f8 Merge pull request #139567 from serathius/watchcache-benchmark-not-older-than
HEAD is now at 359baf1d4d1 cpu_manager: drop support for V1 checkpoint format
[root@raji-x86-workspace1 kubernetes]# gotip test k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state -count=1 -run TestCheckpointStateRestore
--- FAIL: TestCheckpointStateRestore (0.28s)
    --- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=false (0.14s)
        feature_gate.go:170: I0612 10:11:15.255359] Updated featureGates={"PodLevelResourceManagers":false}
        --- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=false/Fail_to_restore_checkpoint_from_v1_(migration) (0.01s)
            state_mem.go:45: I0612 10:11:15.359737] CPUManager state checkpoint/CPUManager state memory: Initialized
            state_checkpoint.go:162: I0612 10:11:15.359906] CPUManager state checkpoint: trying to load v4 CPU manager checkpoint
            state_checkpoint.go:179: I0612 10:11:15.360413] CPUManager state checkpoint: could not load v4 checkpoint, falling back to v3 err="failed to deserialize cpu manager checkpoint: json: cannot unmarshal string into Go struct field CPUManagerCheckpoint.entries.containerID1 of type map[string]string"
            state_checkpoint.go:199: I0612 10:11:15.360482] CPUManager state checkpoint: trying to load v3 CPU manager checkpoint
            state_checkpoint.go:210: I0612 10:11:15.360758] CPUManager state checkpoint: could not load v3 checkpoint, falling back to v2 err="json: cannot unmarshal string into Go struct field CPUManagerCheckpointV3.entries.containerID1 of type map[string]string"
            state_checkpoint.go:231: I0612 10:11:15.360816] CPUManager state checkpoint: trying to load v2 cpu manager checkpoint
            state_checkpoint_test.go:532: 
                	Error Trace:	/root/kubernetes/pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go:532
                	Error:      	Error "could not restore state from checkpoint: json: cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries.containerID1 of type map[string]string, please drain this node and delete the CPU manager checkpoint file \"/tmp/cpumanager_state_test1780922716/cpumanager_checkpoint_test\" before restarting Kubelet" does not contain "cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries of type map[string]string"
                	Test:       	TestCheckpointStateRestore/PodLevelResourceManagers=false/Fail_to_restore_checkpoint_from_v1_(migration)
        feature_gate.go:170: I0612 10:11:15.387715] Updated featureGates={"PodLevelResourceManagers":false}
    --- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=true (0.14s)
        feature_gate.go:170: I0612 10:11:15.388560] Updated featureGates={"PodLevelResourceManagers":true}
        --- FAIL: TestCheckpointStateRestore/PodLevelResourceManagers=true/Fail_to_restore_checkpoint_from_v1_(migration) (0.01s)
            state_mem.go:45: I0612 10:11:15.495815] CPUManager state checkpoint/CPUManager state memory: Initialized
            state_checkpoint.go:162: I0612 10:11:15.495963] CPUManager state checkpoint: trying to load v4 CPU manager checkpoint
            state_checkpoint.go:179: I0612 10:11:15.496218] CPUManager state checkpoint: could not load v4 checkpoint, falling back to v3 err="failed to deserialize cpu manager checkpoint: json: cannot unmarshal string into Go struct field CPUManagerCheckpoint.entries.containerID1 of type map[string]string"
            state_checkpoint.go:199: I0612 10:11:15.496269] CPUManager state checkpoint: trying to load v3 CPU manager checkpoint
            state_checkpoint.go:210: I0612 10:11:15.496492] CPUManager state checkpoint: could not load v3 checkpoint, falling back to v2 err="json: cannot unmarshal string into Go struct field CPUManagerCheckpointV3.entries.containerID1 of type map[string]string"
            state_checkpoint.go:231: I0612 10:11:15.496540] CPUManager state checkpoint: trying to load v2 cpu manager checkpoint
            state_checkpoint_test.go:532: 
                	Error Trace:	/root/kubernetes/pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go:532
                	Error:      	Error "could not restore state from checkpoint: json: cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries.containerID1 of type map[string]string, please drain this node and delete the CPU manager checkpoint file \"/tmp/cpumanager_state_test1780922716/cpumanager_checkpoint_test\" before restarting Kubelet" does not contain "cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries of type map[string]string"
                	Test:       	TestCheckpointStateRestore/PodLevelResourceManagers=true/Fail_to_restore_checkpoint_from_v1_(migration)
        feature_gate.go:170: I0612 10:11:15.524448] Updated featureGates={"PodLevelResourceManagers":false}
FAIL
FAIL	k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state	0.313s
FAIL
[root@raji-x86-workspace1 kubernetes]# go test k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state -count=1 -run TestCheckpointStateRestore
ok  	k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state	0.360s
[root@raji-x86-workspace1 kubernetes]#
```

### Relevant SIG(s)

/sig node

--- a/pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go
+++ b/pkg/kubelet/cm/cpumanager/state/state_checkpoint_test.go
@@ -323,7 +323,7 @@ func TestCheckpointStateRestore(t *testing.T) {
 			}`,
 			"static",
 			containermap.ContainerMap{},
-			"cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries of type map[string]string",
+			"cannot unmarshal string into Go struct field CPUManagerCheckpointV2.entries",
 			nil,
 		},
 		{

